### PR TITLE
Feat: add new OIDC credentials creation and assigment (AWS data source)

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of #743 

1. Added data source.
2. Added required API for oidc sub.
3. Added unit tests for API call.
4. Added acceptance tests for new data source.
5. Added integration tests for new data source. --- removed --- flaky due to oicd policy enablement.
6. Added examples.

### CODE REVIEW

**Comment by tomer-landesman** [comment]
[Init Remote Apply step is not looking for the right flag](https://trello.com/c/DrX6cSWj/11478-init-remote-apply-step-is-not-looking-for-the-right-flag)

File: client/api_key.go [hunk 1]
```diff
@@ -63,3 +63,17 @@ func (client *ApiClient) ApiKeys() ([]ApiKey, error) {
...
 }
+
+func (client *ApiClient) OidcSub() (string, error) {
+	organizationId, err := client.OrganizationId()
+	if err != nil {
+		return "", err
+	}
+
+	var result string
+	if err := client.http.Get("/api-keys/oidc-sub", map[string]string{"organizationId": organizationId}, &result); err != nil {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply client/api_key.go:74]: The data resource returns the oidc sub.
If it's not required, I can remove it.

File: env0/data_aws_oidc_credentials.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+			"id": {
+				Type:         schema.TypeString,
+				Description:  "the id of the aws oidc credentials",
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+			"oidc_sub": {
```
Comment (TomerHeber) [inline env0/data_aws_oidc_credentials.go:28]: @Yossi-kerner - if it's not needed. please let me know...

**Comment by TomerHeber** [comment]
> All in all looks good left some questions. Also I need some more explanation about the last 3 tests in `env0/data_aws_oidc_credentials_test.go`

@Yossi-kerner added some comments and responded to your questions. Please let me know if further information is required. Thank you!

File: env0/data_aws_oidc_credentials_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	})
+
+	t.Run("throw error when no name or id is supplied", func(t *testing.T) {
+		runUnitTest(t,
+			getErrorTestCase(map[string]interface{}{}, "one of `id,name` must be specified"),
+			func(mock *client.MockApiClientInterface) {},
+		)
+	})
+
+	t.Run("throw error when by name and more than one is returned", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Yossi-kerner) [reply env0/data_aws_oidc_credentials_test.go:102]: oh ok I see that now, missed that thanks

**Pull request merged**